lib/cfg.py

```diff
@@ -255,66 +255,59 @@ def cfg_version(filename):
 
 def cfg_altiumWorkspace(options):
     """
+    options è una variabile di tipo <class 'argparse.Namespace' che contiene:
+        1. il path del file Workspace
+        2. se i file da mergiare sono di tipo csv o xlsx
+    
     ricerca del nome di tutti i progetti all'interno del file Workspace
     """
-    risultato=True
     path_dict = {}
     config = ConfigParser.RawConfigParser()
     config.read(options.ws)
-
     for i in config.sections():
         try:
+            #es. temp='nomeprogetto/nomeprogetto.txt' 
             temp = config.get(i, 'ProjectPath')
+            p = temp
             temp = temp.split('/')
-            k = []
-            p = []
+            k = ''
+            k = os.path.join(k, *temp[:len(temp)-1])
 
-            if len(temp) > 1:
-                k = os.path.splitext(temp[1])[0]
-                p = os.path.join(*temp)
-            else:
-                k = os.path.splitext(temp[0])[0]
-                p = os.path.join("./", *temp)
-
-            if len(k) == 0 or len(p) == 0:
-                lib.error("Errore nel parsing del workspace: %s" % e,
-                            self.handler, terminal=self.terminal)
-                risultato =False
-
-            k = k.lower()
+            #es. path_dict = {nomeprogetto : nomeprogetto/nomeprogetto.txt}
             path_dict[k] = p
 
         except ConfigParser.NoOptionError:
             pass
-    path=''
-    ws=options.ws.split('/')
-    for i in range(0, (len(ws)-1)):
-        path=os.path.join(path, ws[i])
+    
+    #calcolo path dove si trovano i progetti
+    ws = options.ws.split('/')
+    path_proj = ''
+    path_proj = os.path.join(path_proj, *ws[:len(ws)-1])
 
     """
-    ricerca parametri per ogni progetto e creazione dizionario con {parametro: valore}
-    ricerca e verifica esistenza file a cui fare il mergebom e messi in una lista
+    ricerca parametri per ogni progetto e esistenza dei file a cui fare il mergebom
     """
-    parametri_dict={}
-    file_BOM=[]
-    if risultato:
-        nprogetti=0 
-        for k, v in path_dict.items():
-
-            prj=os.path.join(path,v)
-            f=open(prj,'r')
-            config=ConfigParser.RawConfigParser()
+    parametri_dict = {}
+    file_BOM = []
```

Comment:
Queste due dichiarazioni di variabili le metterei dentro al `for`, in questo modo non devi re-inizializzarle tutte le volte a fine del ciclo.